### ISSUE TITLE
Error, getting an error that makes VS-Code slow down and become less responsive

### ISSUE THREAD

**Issue by altimmons** [satokaz/vscode-markdown-header-coloring#12]
I am pretty sure the error is with this extension.  I tried disabling them one by one, then re-enabled them sequentially.  It's not as bad with just this extension enabled, but it only occurs with that situation.  I will keep trying to pin it down to make sure.  

I am working on trying to figure out how to find out what is going on.  What happens is that when I type something- for some reason, especially if I hit backspace- it seems to trigger a redraw or a save- but VS Code stops responding for a while, then all the keys (or rather some of them) when it was frozen, all rush out- usually incorrectly.  That of course means I have to backspace again, which starts this all over.  It makes it very difficult to use.  

I should note that my other suspicion is that its a problem with git.

Its hard to figure out what VS code is doing during that time, but the error I get is this:

 ```
 ERR Cannot read property 'document' of undefined: TypeError: Cannot read property 'document' of undefined
	at vscode.window.onDidChangeVisibleTextEditors.e (C:\Users\s163837\.vscode\extensions\satokaz.vscode-markdown-header-coloring-0.1.1\out\extension.js:11:44)
	at e.fire (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:133:218)
	at c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:634:683
	at e.fire (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:133:218)
	at e.$acceptDocumentsAndEditorsDelta (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:541:737)
	at t._doInvokeHandler (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:658:837)
	at t._invokeHandler (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:658:509)
	at t._receiveRequest (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:657:126)
	at t._receiveOneMessage (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:656:59)
	at c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:653:895
	at c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:130:793
	at e.fire (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:133:218)
	at a (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:186:147)
	at Socket.n._socketDataListener (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:186:368)
	at emitOne (events.js:116:13)
	at Socket.emit (events.js:211:7)
	at addChunk (_stream_readable.js:263:12)
	at readableAddChunk (_stream_readable.js:250:11)
	at Socket.Readable.push (_stream_readable.js:208:10)
	at Pipe.onread (net.js:594:20)
```

it is also possible that this error is incidental, and has nothing to do with the interface responsiveness.  

I have gotten used to using the extension and I love it.  I even learned some .css real quick to customize it.  I really hate having it disabled.  I use markdown a ton, and I hate not being able to see the section breaks!

I will keep poking about to see if I can find the problem.  I also should note that I have 3 computers I use.  They all have very similar configurations- though not exactly- and it works on the other two.

If you have any idea to trace whats going on- let me know.

**Comment by satokaz** [comment]
@altimmons 

Thank you for the report.

I may be misunderstanding how to use onDidChangeVisibleTextEditors.
Please wait, I will create an extension that removed onDidChangeVisibleTextEditors.

**Comment by satokaz** [comment]
@altimmons 

Extension that removed onDidChangeVisibleTextEditors.
Could you check whether you reproduce it?

[vscode-markdown-header-coloring-0.1.2.vsix.zip](https://github.com/satokaz/vscode-markdown-header-coloring/files/2867429/vscode-markdown-header-coloring-0.1.2.vsix.zip)

**Comment by satokaz** [comment]
@altimmons 

I published a new 0.1.2 to the marketplace.

**Comment by altimmons** [comment]
Great, problem seems solved.
